service/metrics/metrics.go

```diff
@@ -297,3 +297,24 @@ func (c *measuredConn) ReadFrom(r io.Reader) (int64, error) {
 func MeasureConn(conn onet.DuplexConn, bytesSent, bytesReceived *int64) onet.DuplexConn {
 	return &measuredConn{DuplexConn: conn, writeCount: bytesSent, readCount: bytesReceived}
 }
+
+// NoOpMetrics is a fake ShadowsocksMetrics that doesn't do anything. Useful in tests
+// or if you don't want to track metrics.
+type NoOpMetrics struct{}
+
+func (m *NoOpMetrics) SetBuildInfo(version string) {}
+func (m *NoOpMetrics) AddTCPProbe(clientLocation, status, drainResult string, port int, data ProxyMetrics) {
```

Comment:
Nit: `{}` vs `{\n}` consistency.

Reply:
I can't. That's the auto formatter. It reverts when I save.

Reply:
OK